internal/userutils/userutils_test.go

```diff
@@ -0,0 +1,126 @@
+	} else if canUseSudoNonInteractively() {
+		log.Info(context.Background(), "Can use sudo non-interactively")
+		useSudo = true
+	} else {
+		t.Skip("Skipping test: requires root privileges or unprivileged user namespaces")
+	}
+
+	// Build the helper binary
+	tempDir := t.TempDir()
+	gpasswdHelperPath = tempDir + "/gpasswd"
+	//nolint:gosec // G204 It's fine to pass a variable to exec.Command here
+	cmd := exec.Command("go", "build",
+		"-o", gpasswdHelperPath,
+		"-tags", "userutils_testhelpers",
+		"./testhelpers/gpasswd.go")
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	err := cmd.Run()
+	require.NoError(t, err)
+
+	// Create a temporary group file
+	groupFile = t.TempDir() + "/group"
+	require.NoError(t, err)
+	// Create the temporary group file
+	//nolint:gosec // G306 The group file is expected to have permissions 0644
+	err = os.WriteFile(groupFile, []byte("testgroup:x:1001:testuser"), 0644)
+	require.NoError(t, err)
+
+	// Set the group file to the temporary file
+	origGroupFile := userutils.GroupFile
+	userutils.GroupFile = groupFile
+	defer func() { userutils.GroupFile = origGroupFile }()
+
+	// Try using gpasswd to modify the group file. This should succeed, because
+	// the group file is not locked.
+	output, err := runGPasswd("--add", "root", "testgroup")
+	require.NoError(t, err, string(output))
+
+	// Lock the group file
+	err = userutils.LockGroupFile()
+	require.NoError(t, err)
+
+	// Try using gpasswd to modify the group file. This should fail, because
+	// the group file is locked.
+	output, err = runGPasswd("--delete", "root", "testgroup")
+	require.Error(t, err, string(output))
+	require.Contains(t, string(output), "gpasswd: cannot lock /etc/group")
+
+	// Try locking the group file again. This should fail, because the group file is already locked.
+	err = userutils.LockGroupFile()
+	require.Error(t, err)
+
+	// Unlock the group file
+	err = userutils.UnlockGroupFile()
+	require.NoError(t, err)
+
+	// Try using gpasswd to modify the group file again. This should succeed,
+	// because the group file is unlocked.
+	output, err = runGPasswd("--delete", "root", "testgroup")
+	require.NoError(t, err, string(output))
+}
+
+func canUseUnprivilegedUserNamespaces() bool {
+	cmd := exec.Command("bwrap", "--ro-bind", "/", "/", "--unshare-user", "--uid", "0", "/bin/true")
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	if err := cmd.Run(); err != nil {
+		log.Warningf(context.Background(), "Can't use unprivileged user namespaces: %v", err)
+		return false
+	}
+
+	return true
+}
+
+func canUseSudoNonInteractively() bool {
+	cmd := exec.Command("sudo", "-Nnv")
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	if err := cmd.Run(); err != nil {
+		log.Warningf(context.Background(), "Can't use sudo non-interactively: %v", err)
+		return false
+	}
+
+	return true
+}
+
+func runGPasswd(args ...string) ([]byte, error) {
+	args = append([]string{
+		"env", "GROUP_FILE=" + groupFile,
```

Comment:
```suggestion
		"env", "LANG=C", "GROUP_FILE=" + groupFile,
```

Since we're parsing the output, so we need to make sure gettext isn't affecting us

Reply:
great catch!